core/model/commands/add.hpp

```diff
@@ -15,56 +15,56 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-#ifndef CORE_DOMAIN_TRANSACTIONS_TRANSFERTRANSACTION_HPP_
-#define CORE_DOMAIN_TRANSACTIONS_TRANSFERTRANSACTION_HPP_
-
-#include "transaction.hpp"
-#include "command.hpp"
-#include "objects.hpp"
+#ifndef CORE_DOMAIN_COMMANDS_ADD_HPP_
+#define CORE_DOMAIN_COMMANDS_ADD_HPP_
 
 #include "../../service/json_parse.hpp"
 
+#include "../objects/domain.hpp"
+#include "../objects/asset.hpp"
+#include "../objects/message.hpp"
+
 #include <string>
-#include <type_traits>
 
 namespace command {
 
-template<typename T, std::enable_if_t<std::is_base_of<AbsObject, T>::value,std::nullptr_t> = nullptr>
-class Add : public Command{
-    T object;
+template <typename T>
+class Add: public T {
   public:
-    Add(T object) :
-        object(object)
-    {}
 
-    std::string getAsJson() const{
-        object.getAsJson();
-    }
+    using Object = json_parse::Object;
+    using Rule = json_parse::Rule;
+    using Type = json_parse::Type;
+
+    Add(
+        const std::string& domain,
+        const std::string& name,
+        const unsigned long long& value,
+        const unsigned int& precision
+    );
+
+    Add(const std::string& ownerPublicKey,const std::string& name);
+
+    Add(Object obj);
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Class 'Add' has a constructor with 1 argument that is not explicit.](https://www.codacy.com/app/soramitsu/iroha/file/4478128047/issues/source?bid=3712491&fileBranchId=3923596#l48)